File: leo2-central/src/main/java/org/deku/leo2/central/rest/services/internal/v1/TestService.java
package org.deku.leo2.central.rest.services.internal.v1;

import org.deku.leo2.rest.entities.internal.v1.TestEntry;

import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;
import java.util.ArrayList;

/**
 * Created by masc on 23.07.14.
 */
@Path("internal/v1/test")
@Produces(MediaType.APPLICATION_JSON)
public class TestService implements org.deku.leo2.rest.services.internal.v1.TestService {
    @Override
    public TestEntry[] get() {
        ArrayList<TestEntry> entries = new ArrayList<>();
        for (int i = 0; i < 50; i++)
            entries.add(new TestEntry("test"));
        return entries.toArray(new TestEntry[0]);
    }

    @Override
    public TestEntry get(String name) {
        return null;
    }

    @Override
    public TestEntry[] find(String query) {
        return new TestEntry[0];
    }
}

File: leoz-common/src/main/java/org/deku/leoz/service/internal/TestService.kt
package org.deku.leoz.service.internal

import io.swagger.annotations.Api
import io.swagger.annotations.ApiOperation
import sx.rs.auth.ApiKey
import javax.ws.rs.*
import javax.ws.rs.core.MediaType
import javax.ws.rs.core.Response

/**
 * Created by masc on 09.10.15.
 */
@Path("internal/v1/test")
@Produces(MediaType.APPLICATION_JSON)
@Consumes(MediaType.APPLICATION_JSON)
@Api(value = "Test operations")
@ApiKey(false)
interface TestService {
    /**
     * Get entry by name
     * @param name name
     */
    @GET
    @Path("/test-soap-call")
    @ApiOperation(value = "Invoke soap call for testing")
    fun testSoapCall(): Response

    @PATCH
    @Path("/test-publish-updateinfo-mobile")
    fun testPublishUpdateInfoToMobile()
}